stereo_image_proc/src/stereo_image_proc/disparity_fpga.cpp

```diff
@@ -0,0 +1,399 @@
+    }
+    else {
+      continue;
+    } // end-if-else-if
+
+
+  } // end-for
+
+  pub_disparity_= this->create_publisher<stereo_msgs::msg::DisparityImage>("disparity", 1);
+
+  connectCb();
+}
+
+// Handles (un)subscribing when clients (un)subscribe
+void DisparityNodeFPGA::connectCb()
+{
+  // TODO(jacobperron): Add unsubscribe logic when we use graph events
+  image_transport::TransportHints hints(this, "raw");
+  const bool use_system_default_qos = this->get_parameter("use_system_default_qos").as_bool();
+  rclcpp::QoS image_sub_qos = rclcpp::SensorDataQoS();
+  if (use_system_default_qos) {
+    image_sub_qos = rclcpp::SystemDefaultsQoS();
+  }
+  const auto image_sub_rmw_qos = image_sub_qos.get_rmw_qos_profile();
+  sub_l_image_.subscribe(this, "left/image_rect", hints.getTransport(), image_sub_rmw_qos);
+  sub_l_info_.subscribe(this, "left/camera_info", image_sub_rmw_qos);
+  sub_r_image_.subscribe(this, "right/image_rect", hints.getTransport(), image_sub_rmw_qos);
+  sub_r_info_.subscribe(this, "right/camera_info", image_sub_rmw_qos);
+}
+
+void DisparityNodeFPGA::imageCb(
+  const sensor_msgs::msg::Image::ConstSharedPtr & l_image_msg,
+  const sensor_msgs::msg::CameraInfo::ConstSharedPtr & l_info_msg,
+  const sensor_msgs::msg::Image::ConstSharedPtr & r_image_msg,
+  const sensor_msgs::msg::CameraInfo::ConstSharedPtr & r_info_msg)
+{
+  // If there are no subscriptions for the disparity image, do nothing
+  if (pub_disparity_->get_subscription_count() == 0u) {
+    return;
+  }
+
+  // Update the camera model
+  model_.fromCameraInfo(l_info_msg, r_info_msg);
+
+  // Allocate new disparity image message
+  auto disp_msg = std::make_shared<stereo_msgs::msg::DisparityImage>();
+  disp_msg->header = l_info_msg->header;
+  disp_msg->image.header = l_info_msg->header;
+
+  // Allocate a new disparity info message
+  sensor_msgs::msg::CameraInfo::ConstSharedPtr disp_info_msg = std::make_shared<sensor_msgs::msg::CameraInfo>(*l_info_msg);
+
+  // Compute window of (potentially) valid disparities
+  // int border = block_matcher_.getCorrelationWindowSize() / 2;
+  // int left = block_matcher_.getDisparityRange() + block_matcher_.getMinDisparity() + border - 1;
+  // int wtf;
+  /* if (block_matcher_.getMinDisparity() >= 0) {
+     wtf = border + block_matcher_.getMinDisparity();
+  } else {
+     wtf = std::max(border, -block_matcher_.getMinDisparity());
+  } */
+  // int right = disp_msg->image.width - 1 - wtf;
+  // int top = border;
+  // int bottom = disp_msg->image.height - 1 - border;
+  // disp_msg->valid_window.x_offset = left;
+  // disp_msg->valid_window.y_offset = top;
+  // disp_msg->valid_window.width = right - left;
+  // disp_msg->valid_window.height = bottom - top;
+
+  // Vitis Vision library ROS to CV image
+  cv_bridge::CvImagePtr cv_ptr_l;
+  cv_bridge::CvImagePtr cv_ptr_r;
+  cv::Mat hls_disp16;
+  static const double inv_dpp = 1.0 / NO_OF_DISPARITIES;
```

Comment:
I would use constexp double 
anyways, vitis stereo_lbm uses 12.4 fixed point, you have to the divide by 16, no ?